core/events/events_procedure_change_return.ts

```diff
@@ -5,19 +5,39 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import {IProcedureModel} from '../interfaces/i_procedure_model.js';
+import type {IProcedureModel} from '../interfaces/i_procedure_model.js';
 import * as registry from '../registry.js';
-import {Workspace} from '../workspace.js';
+import type {Workspace} from '../workspace.js';
 
 import {ProcedureBase} from './events_procedure_base.js';
 import * as eventUtils from './utils.js';
 
 
 export class ProcedureChangeReturn extends ProcedureBase {
+  type = eventUtils.PROCEDURE_CHANGE_RETURN;
+  private newTypes: string[]|null;
+
   constructor(
       workpace: Workspace, model: IProcedureModel,
       public readonly oldTypes: string[]|null) {
     super(workpace, model);
+
+    this.newTypes = model.getReturnTypes();
+  }
+
+  run(forward: boolean) {
```

Comment:
Is it the convention that runnable events have a `forward` parameter, rather than `backward`?  I'd have thought that the default (`undefined` being falsey) should be to run the event forward…

Reply:
Yep, it's defined on the abstract class: https://github.com/google/blockly/blob/485bd4dddf24f3c36db49565de10d8c0b0a3ba5d/core/events/events_abstract.ts#L115